crates/turborepo-lib/src/commands/login.rs

```diff
@@ -0,0 +1,106 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::{anyhow, Result};
+use axum::{extract::Query, response::Redirect, routing::get, Router};
+use log::{debug, warn};
+use serde::Deserialize;
+use tokio::sync::OnceCell;
+
+use crate::{
+    client::{APIClient, UserClient},
+    commands::CommandBase,
+    config::{default_user_config_path, UserConfigLoader},
+    get_version,
+    ui::{BOLD, CYAN},
+};
+
+const DEFAULT_HOST_NAME: &str = "127.0.0.1";
+const DEFAULT_PORT: u16 = 9789;
+
+pub async fn login(base: CommandBase) -> Result<()> {
+    let repo_config = base.repo_config()?;
+    let login_url_base = repo_config.login_url();
+    debug!("turbo v{}", get_version());
+    debug!("api url: {}", repo_config.api_url());
+    debug!("login url: {login_url_base}");
+
+    let redirect_url = format!("http://{DEFAULT_HOST_NAME}:{DEFAULT_PORT}");
+    let login_url = format!("{login_url_base}/turborepo/token?redirect_uri={redirect_url}");
+    println!(">>> Opening browser to {login_url}");
+
```

Comment:
I noticed we're not having a spinner displaying `Waiting for your authorization...` in the console anymore. Is there something preventing us from using [`indicatif`](https://docs.rs/indicatif/latest/indicatif/index.html) to keep this functionality? I've had good luck with the spinner and custom tick strings.

Reply:
Ah good catch. Meant to add a spinner library